src/inference/gibbs.jl

```diff
@@ -83,11 +88,72 @@ metadata(t::GibbsTransition) = (lp = t.lp,)
 DynamicPPL.getlogp(t::GibbsTransition) = t.lp
 
 # extract varinfo object from state
-getvarinfo(state) = state.vi
-getvarinfo(state::AbstractVarInfo) = state
+"""
+    gibbs_varinfo(model, sampler, state)
+
+Return the variables corresponding to the current `state` of the Gibbs component `sampler`.
+"""
+gibbs_varinfo(model, sampler, state) = varinfo(state)
+varinfo(state) = state.vi
```

Comment:
Maybe consider introducing an `AbstractMCMCState` type to avoid potential misuses of `varinfo`? It also improves clarity I think. In simple cases, samplers can use a generic `MCMCState`, wherein more complicated ones, a sampler-specific `MCMCState` such as `HMCState` can be defined.